### PR TITLE
Sam.jaberi

### PR DESCRIPTION
**Pull request by samnjab**

### CODE REVIEW

File: src/main/java/quotify_app/app/AppBuilder.java [hunk 1]
```diff
@@ -80,6 +83,24 @@ public AppBuilder addLoginUseCase() {
...
+    public AppBuilder addLandingView() {
+        cardPanel.add(landingFactory.getLandingView(), "landing"); // Add Landing View
+        return this;
+    }
+
+    /**
+     * Adds the Landing Use Case to the application.
+     * @return this builder
+     */
+    public AppBuilder addLandingUseCase() {
```
Comment (ranb0m) [inline src/main/java/quotify_app/app/AppBuilder.java:104]: note: adding controller twice?

File: src/main/java/quotify_app/app/factories/LandingFactory.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    /**
+     * Sets up the LandingController.
+     * @param viewManagerModel The ViewManagerModel from the AppBuilder.
+     */
+    public void setUpController(ViewManagerModel viewManagerModel) {
+        // Instantiate AreaStore for managing cached areas
+        final AreaStore areaStore = new AreaStore();
+
+        // Instantiate AreaDataAccessObject with the AreaStore
+        final AreaDataAccessObject areaDataAccessObject = new AreaDataAccessObject(areaStore);
```
Comment (ranb0m) [inline src/main/java/quotify_app/app/factories/LandingFactory.java]: I believe both the areadataaccessobjects and the areastore should be instantiated in the app builder, similar to how the other dataaccessobjects are being created

File: src/main/java/quotify_app/app/factories/LandingFactory.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        this.landingView = new LandingView(landingViewModel);
+    }
+
+    /**
+     * Sets up the LandingController.
+     * @param viewManagerModel The ViewManagerModel from the AppBuilder.
+     */
+    public void setUpController(ViewManagerModel viewManagerModel) {
+        // Instantiate AreaStore for managing cached areas
+        final AreaStore areaStore = new AreaStore();
```
Comment (ranb0m) [inline src/main/java/quotify_app/app/factories/LandingFactory.java]: should this maybe be instantiated in the appbuilder and set as a class var?

File: src/main/java/quotify_app/app/factories/LandingFactory.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        final AreaDataAccessObject areaDataAccessObject = new AreaDataAccessObject(areaStore);
+
+        // Setup Presenter and Interactor for Landing with necessary dependencies
+        final LandingOutputBoundary landingPresenter = new LandingPresenter(
+                landingViewModel,
+                viewManagerModel
+        );
+
+        final LandingInputBoundary landingInteractor = new LandingInteractor(
+                areaDataAccessObject, new PropertyDataAccessObject(), landingPresenter);
```
Comment (ranb0m) [inline src/main/java/quotify_app/app/factories/LandingFactory.java]: same here, the dataaccessobjects should be instantiated in the appbuilder

File: src/main/java/quotify_app/app/AppBuilder.java [hunk 5]
```diff
@@ -80,6 +83,24 @@ public AppBuilder addLoginUseCase() {
...
+        landingFactory.setUpController(viewManagerModel);
```
Comment (ranb0m) [inline src/main/java/quotify_app/app/AppBuilder.java]: follow the formatting that the other addusecase functions use, it should work for you here

File: src/main/java/quotify_app/app/AppBuilder.java [hunk 6]
```diff
@@ -42,6 +44,7 @@ public AppBuilder() {
         comparatorFactory.setUpController(viewManagerModel);
         currentPriceFactory.setUpController(viewManagerModel);
         futurePriceFactory.setUpController(viewManagerModel);
+        landingFactory.setUpController(viewManagerModel);
```
Comment (ranb0m) [inline src/main/java/quotify_app/app/AppBuilder.java]: this is where you'd be passing in whatever necessary arguments you need for setting up the controller: i.e. the areadataaccessobject, propertydataaccessobject, areacache, along iwth the viewmanagermodel

Just instantiate the above 3 with the other class variables

**Review by ranb0m — CHANGES_REQUESTED** [review]

File: src/main/java/quotify_app/app/factories/LandingFactory.java (hunk 3 above)
Comment (samnjab) [reply src/main/java/quotify_app/app/factories/LandingFactory.java]: This is now instantiated in appBuilder, not as a global but accessible through areaAccessObject.getCache()

File: src/main/java/quotify_app/app/AppBuilder.java (hunk 1 above)
Comment (samnjab) [reply src/main/java/quotify_app/app/AppBuilder.java:104]: Is this still an issue? 
`public AppBuilder addLandingUseCase() {
        landingFactory.getLandingView().setLandingController(landingFactory.getLandingController());
        return this;
    }`

Comment (ranb0m) [reply src/main/java/quotify_app/app/AppBuilder.java:104]: this one should be good like this

File: src/main/java/quotify_app/app/factories/LandingFactory.java (hunk 2 above)
Comment (samnjab) [reply src/main/java/quotify_app/app/factories/LandingFactory.java]: this is done

File: src/main/java/quotify_app/app/factories/LandingFactory.java (hunk 3 above)
Comment (samnjab) [reply src/main/java/quotify_app/app/factories/LandingFactory.java]: this is done

File: src/main/java/quotify_app/app/factories/LandingFactory.java (hunk 4 above)
Comment (samnjab) [reply src/main/java/quotify_app/app/factories/LandingFactory.java]: done

File: src/main/java/quotify_app/app/AppBuilder.java (hunk 6 above)
Comment (samnjab) [reply src/main/java/quotify_app/app/AppBuilder.java]: done

**Pull request merged**